java/src/org/openqa/selenium/remote/RemoteWebDriver.java

```diff
@@ -815,9 +817,9 @@ public String toString() {
       platformName = "unknown";
     }
 
-    return String.format(
-        "%s: %s on %s (%s)",
-        getClass().getSimpleName(), caps.getBrowserName(), platformName, getSessionId());
+      return new Formatter(Locale.US).format(
```

Comment:
It might be a good idea to ensure a single `Formatter` instance is used and that instance is closed too.